syntax = "proto3";

option go_package = "ConnectionStats/connstatsprotobuf";

package connstatsprotobuf;

// The greeting service definition.
service StatsService {
  // Sends a connection stats
  rpc CollectStats (StatsRequest) returns (StatsReply) {}

}

message ConnectionStat {
    string a_ip  = 1;       //netip.Addr
	string b_ip  = 2;       //netip.Addr
	uint32 a_port = 3;      
	uint32 b_port = 4;      
	uint64 packets_in = 5;  
	uint64 packets_out = 6;
	uint64 ts_ini = 7;     
	uint64 ts_fin = 8;     
	uint64 bytes_in = 9;   
	uint64 bytes_out = 10;  
  }

// The request message.
message StatsRequest {
  
}

// The response message containing the stats table
message StatsReply {
    repeated ConnectionStat connstat = 1;
}